src/Raven.Server/Documents/Queries/Graph/QueryQueryStep.cs

```diff
@@ -114,13 +114,20 @@ public ValueTask Initialize()
         {
             if (_index != -1)
                 return default;
-
-            var results = _queryRunner.ExecuteQuery(new IndexQueryServerSide(_queryMetadata)
+            //TODO: we can cache only queries we know has identical steps to save memory 
+            //TODO: we can also add ref count per step and clear the results once they are populated
+            var cache = _graphQueryPlan.QueryCache;
+            var key = _queryMetadata.Query.ToString();
```

Comment:
I don't particularly like this, because of the allocations. But we can live with this for now

Reply:
At this point, this isn't worth it, I agree. We can wait for the re-impl